cmd/grumble/server.go

```diff
@@ -1461,7 +1504,12 @@ func (server *Server) Stop() (err error) {
 		client.Disconnect()
 	}
 
-	// Close the TLS listener and the TCP listener
+	// Wait for the HTTP server to shutdown gracefully
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(15*time.Second))
+	server.webhttp.Shutdown(ctx)
+	cancel()
```

Comment:
Why cancel after Shutdown has returned?

Reply:
This is recommended behaviour by the context package. It's done in order to prevent the timer from leaking (you'll also get a `go vet` error if you don't call the cancel function).

Reply:
Thanks, I've not done much Go after contexts were introduced.